File: src/main/java/InnerClass.java
/**
 * Created by WenTaoKing on 2019-06-11
 * description:
 */
public class InnerClass {
    class nonStaticInner{
        public void print(){
            System.out.println("我是非静态内部类");
        }
    }

    public static class StaticInner{
        public static void print(){
            System.out.println("我是静态内部类");
        }
    }

    public static void main(String[] args){
        InnerClass.StaticInner.print();                  //静态内部类

        new InnerClass().new nonStaticInner().print();  //非静态内部类
    }
}

File: src/main/kotlin/innerclass/InnerClass.kt
package innerclass

/**
 * Created by wt.King on 2019-06-10
 * description:
 */


//Kotlin里内部类默认是静态内部类，用inner表示非静态内部类
class Outter {
    inner class Inner {
        fun print() {
            println("我是非静态内部类")
        }
    }

    class StaticInner {
        fun print2() {
            println("我是静态内部类")
        }
    }
}

fun main() {

    println( Outter().Inner().print() )
    println(Outter.StaticInner().print2())
}